dxp/liferay_global_event_handler.md

```diff
@@ -0,0 +1,213 @@
+# Liferay’s Global Event Handler
+
+Liferay’s event handler helps the developers for creating/listening/firing/removing custom events that can be listened in whole DXP which is rely on the YUI3 event target implementation. See the implementation [here](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L120).
+
+### First Considerations
+
+First, just consider using the global event handler if necessary. Try to use the default event listener/[callback pattern(if React)](https://reactjs.org/docs/lifting-state-up.html#lifting-state-up) when working. Just use the global event handler when trying to communicate between widgets or not accessible contexts(e.g iframes).
+
+### List of Available Functions
+
+Here is a list of available functions from Liferay’s global variable:
+
+`Liferay.on`
+`Liferay.once`
+`Liferay.onceAfter`
+`Liferay.after`
+`Liferay.before`
+`Liferay.detach`
+`Liferay.detachAll`
+`Liferay.publish`
+`Liferay.fire`
+
+### `Liferay.on`
+
+Description from [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L217):
+
+This function subscribes a callback function to a custom event fired by this object or from an object that bubbles its events to this object.
+
+Callback functions for events published with `emitFacade = true` will receive an `EventFacade` as the first argument (typically named “e”). These callbacks can then call `e.preventDefault()` to disable the behaviour published to that event’s `defaultFn` . See the `EventFacade` API for all available properties and methods. Subscribers to non- `emitFacade` events will receive the arguments passed to `fire()` after the event name.
```

Comment:
> (typically named “e”)

That's probably still the case in DXP, but I think we want people to name that `event` because we don't advise using abbreviations

Reply:
> See the `EventFacade` API

Maybe we could add a link to that

Reply:
> but I think we want people to name that event because we don't advise using abbreviations

Yep: https://github.com/liferay/liferay-frontend-guidelines/blob/master/general/naming.md#avoid-abbreviations

Reply:
👍 Since we're avoiding abbreviations on DXP code makes sense remove it :)